### ISSUE TITLE
Installation issue: tau@2.30.1%gcc@7.1.0

### ISSUE THREAD

**Issue by afrah** [LLNL/spack#23243]
I have to profile some software packages like superlu-dist with tau using a  x86_64 GNU/Linux cluster . I installed spack using this command:
git clone https://github.com/spack/spack.git.  The command for tau installation is :

> spack install tau@2.30.1%gcc@7.1.0+fortran+mpi+openmp+papi+pdt+pthreads

The compiler I should use is  gcc@7.1.0. The installation gives error with tau building stage:
 

> ld: fatal error: cannot use --plugin: ld was compiled without plugin support
>  312    collect2: error: ld returned 1 exit status
>  313    make[1]: *** [tau_convert] Error 1
> 

I tried to add -fno-use-linker-plugin to gcc@7.1.0 compiler flags but that does not help.  The specifications of the compiler from is like this

> - compiler:
    paths:
      cc: /x/progs/gcc/GCC-7.1.0/bin/gcc
      cxx: /x/progs/gcc/GCC-7.1.0/bin/g++
      f77: /x/progs/gcc/GCC-7.1.0/bin/gfortran
      fc: /x/progs/gcc/GCC-7.1.0/bin/gfortran
    operating_system: centos7
    target: x86_64
    modules: []
    environment: {}
    extra_rpaths: []
    flags:
      cflags: -fuse-linker-plugin
      cxxflags: -fuse-linker-plugin
      cppflags: -fuse-linker-plugin
      ldflags: -fuse-linker-plugin
    spec: gcc@7.1.0

The output result, the spack-build-env and the spack-build-out are attached.  Using a different compiler like gcc@8.3.0 and gcc@10.2.0 even gives the same error.

Any idea is highly appreciated.
[spack-build-out.txt](https://github.com/spack/spack/files/6371502/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/6371504/spack-build-env.txt)
[spack-build-01-install-out.txt](https://github.com/spack/spack/files/6371505/spack-build-01-install-out.txt)

[output.txt](https://github.com/spack/spack/files/6371507/output.txt)

**Comment by cessenat** [comment]
Before the root of the problem is solved you can retry with
> spack install tau@2.30.1%gcc@7.1.0+fortran+mpi+openmp+papi+pdt+pthreads ^binutils+plugins

I would like binutils to have the 'plugins' variant at the same time as the 'gold' one but I'm not 100% sure.

**Comment by afrah** [comment]
binutils installed successfully . I used the same command on my computer and tau works perfectly but I need to use the cluster .. from the output file :
[+] /x/tauSpack/spack/opt/spack/linux-centos7-broadwell/gcc-7.1.0/binutils-2.33.1-gc5tqoqftleaovrmwoiqjaey6upkm6zi
I will try anyways

**Comment by cessenat** [comment]
You should not get the message "ld: fatal error: cannot use --plugin: ld was compiled without plugin support"
if you compiled binutils+plugins.  
Can you run spack spec /gc5tqoqf ?

**Comment by afrah** [comment]
I delete the previous installations and run it again with 
`spack install tau@2.30.1%gcc@7.1.0+fortran+mpi+openmp+papi+pdt+pthreads ^binutils+plugins`

I am waiting for the output

**Comment by cessenat** [comment]
For tau you must also set +libiberty+headers (take the binutils spec from tau and add +plugins).

**Comment by afrah** [comment]
Thank you for your reply ... I finally installed tau. I did a set of things besides the command :
`spack install tau@2.30.1%gcc@7.1.0+fortran+mpi+openmp+papi+pdt+pthreads ^binutils+plugins
`
I removed these two lines:
```
module load gcc/7.1.0
module load mpi/openmpi-3.0.0-gcc-7.1.0

```
There was a minor bug in the llibelf header files which requires copying libelf folder inside itself so as as the path libelf/libelf.h exists.
and it works!
